hardinfo/info.c

```diff
@@ -154,10 +181,17 @@ static void flatten_group(GString *output, const struct InfoGroup *group)
     if (group->fields) {
         for (i = 0; i < group->fields->len; i++) {
             struct InfoField field;
+            gchar *tag;
 
             field = g_array_index(group->fields, struct InfoField, i);
 
-            g_string_append_printf(output, "%s=%s\n", field.name, field.value);
+            if (field.mark_for_selection || field.tag) {
+                tag = g_strdup_printf("$%s%s$", field.mark_for_selection ? "*" : "", field.tag ? field.tag : "");
+            } else {
+                tag = g_strdup("");
+            }
+
+            g_string_append_printf(output, "%s%s=%s\n", tag, field.name, field.value);
```

Comment:
`tag` isn't freed here.  This might be better to just do something like:

```C
if (field.mark_for_selection) {
        g_string_append_printf(output, "*");
} else if (field.tag) {
        g_string_append_printf(output, with tag);
} else {
      g_string_append_printf(output, without tag...);
}
```

This way you avoid allocating from the heap unnecessarily.